pkg/operator/defaultscccontroller/syncer.go

```diff
@@ -10,21 +10,39 @@ import (
 )
 
 type Syncer struct {
-	lister   securityv1listers.SecurityContextConstraintsLister
-	recorder events.Recorder
+	lister        securityv1listers.SecurityContextConstraintsLister
+	recorder      events.Recorder
+	updater       OperatorStatusUpdater
+	defaultSCCSet *DefaultSCC
 }
 
 func (s *Syncer) Sync(key types.NamespacedName) error {
-	current, err := s.lister.Get(key.Name)
-	if err != nil {
-		if k8serrors.IsNotFound(err) {
-			klog.Infof("[%s] key=%s scc has been deleted - %s", ControllerName, key.Name, err.Error())
-			return nil
+	// If it's not to do with the default SCC, we don't care.
+	if !s.defaultSCCSet.IsDefault(key.Name) {
+		return nil
+	}
+
+	mutated := make([]string, 0)
+	for _, original := range s.defaultSCCSet.List() {
+		current, err := s.lister.Get(original.GetName())
+		if err != nil {
+			if k8serrors.IsNotFound(err) {
```

Comment:
If a default scc has been removed, is it okay to not set upgradeable=false? I am not sure what the state of a cluster is when a default scc is removed.